collector/monitors/balance_monitor.go

```diff
@@ -64,12 +64,17 @@ func (m *OperatorBotBalanceMonitor) Handler(ctx context.Context) error {
 	coins := resp.GetPayload().Result
 	for _, coin := range coins {
 		if coin.Denom == UUSDDenom {
-			amount, err := strconv.ParseFloat(coin.Amount, 64)
+			amount, err := types.NewDecFromStr(coin.Amount)
 			if err != nil {
-				// if for some reason we cannot parse uusd float amount, just leave previous value
-				return fmt.Errorf("failed to parse coins uusd amount: %w", err)
+				return fmt.Errorf("failed to parse coins uusd amount: %s", coin.Amount)
 			}
-			m.balanceUST.Set(amount / 1_000_000)
+
+			amountFloat, err := amount.Float64()
```

Comment:
![image](https://user-images.githubusercontent.com/6247919/137286412-ad1e8fb7-7f5e-4701-800e-80cacac65174.png)
Seems like this is just a wrapper around strconv